src/jsonldparser.js

```diff
@@ -0,0 +1,97 @@
+import jsonld from 'jsonld'
+
+const rdf = {
+  first: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#first',
+  rest: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#rest',
+  nil: 'http://www.w3.org/1999/02/22-rdf-syntax-ns#nil'
+}
+
+/**
+ * Parses json-ld formatted JS objects to a rdf Term.
+ * @param kb - The DataFactory to use.
+ * @param obj - The json-ld object to process.
+ * @return {Literal|Literal|Literal|NamedNode|NamedNode|NamedNode}
+ */
+export function jsonldObjectToTerm (kb, obj) {
+  if (typeof obj === 'string') {
+    return kb.literal(obj)
+  }
+
+  if (Object.prototype.hasOwnProperty.call(obj, '@list')) {
```

Comment:
This has been added, the end-user can still choose between plain RDF and the $rdf:Collection so that it also solves https://github.com/linkeddata/rdflib.js/issues/169 for json-ld and n3/turtle